### PR TITLE
Rename composer_run_name tag to run_name when logging to MLflow

### PR DESCRIPTION
**Pull request by jerrychen109**
# What does this PR do?
Rename `composer_run_name` tag in MLflow to just `run_name`

image

# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/loggers/mlflow_logger.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def init(self, state: State, logger: Logger) -> None:
                 # Search for an existing run tagged with this Composer run.
                 assert self._experiment_id is not None
                 existing_runs = mlflow.search_runs(experiment_ids=[self._experiment_id],
-                                                   filter_string=f'tags.composer_run_name = "{state.run_name}"',
```
Comment (dakinggg) [inline composer/loggers/mlflow_logger.py:165]: do we need to search for the old key too? probably just to make no workloads get randomly broken?

Comment (jerrychen109) [reply composer/loggers/mlflow_logger.py:165]: probably...backwards compatibility is such a concept

**Review by dakinggg — APPROVED** [review]

**Review by dakinggg — CHANGES_REQUESTED** [review]
until tested

**Comment by jerrychen109** [comment]
@dakinggg Actually added a unit test instead of YOLOing with github unit tests

**Review by dakinggg — APPROVED** [review]

**Pull request merged**